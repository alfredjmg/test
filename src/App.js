import React, { useState } from "react";
import Header from "./components/header/header";
import Content from "./components/content/content";

function App() {
  const [textValue, setTextValue] = useState("");
  // const [isPalindrome, setIsPalindrome] = useState("");
  // const [revertedText, setReverdedText] = useState("");
  const [errorContent, setErrorContent] = useState("");
  const [activeError, setActiveError] = useState(false);

  return (
    <div className="App">
      <div className="container-fluid">
        <Header
          textValue={textValue}
          setTextValue={setTextValue}
          // setIsPalindrome={setIsPalindrome}
          setErrorContent={setErrorContent}
          // setReverdedText={setReverdedText}
          setActiveError={setActiveError}
        />
        <Content
          activeError={activeError}
          errorContent={errorContent}
        // isPalindrome={isPalindrome}
        // revertedText={revertedText}
        />
      </div>
    </div>
  );
}

export default App;
