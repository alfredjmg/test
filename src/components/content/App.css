.content-container {
  margin-top: 3vh;
  display: flex !important;
  text-align: center;
  justify-content: center;
}
.content {
  width: 60%;
  background-color: white;
  padding-top: 5vh;
  padding-bottom: 50vh;
}

.title-content {
  font-size: 1.5vw;
}

.palindrome-text {
  text-align: left;
}

.error-element {
  text-align: left;
  color: red;
}
