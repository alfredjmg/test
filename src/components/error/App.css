body {
  background-color: rgb(199, 199, 199) !important;
}

.input-header {
  background-color: red !important;
  display: flex !important;
  text-align: center;
  justify-content: center;
  vertical-align: middle;
  align-items: center;
  padding: 15px 0;
}

.content-container {
  margin-top: 3vh;
  display: flex !important;
  text-align: center;
  justify-content: center;
}
.content {
  width: 60%;
  background-color: white;
  padding-top: 5vh;
  padding-bottom: 50vh;
}

.title-content {
  font-size: 1.5vw;
}

.palindrome-text {
  text-align: left;
}

.error-element {
  text-align: left;
  color: red;
}

@media screen and (max-width: 768px) {
  .content {
    width: 80%;
  }
  .title-content {
    font-size: 3vw;
    margin-bottom: 2vh;
  }
}

@media screen and (max-width: 576px) {
  .title-content {
    font-size: 4vw;
  }
}
