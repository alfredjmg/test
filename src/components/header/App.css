body {
  background-color: rgb(199, 199, 199) !important;
}

.input-header {
  background-color: red !important;
  display: flex !important;
  text-align: center;
  justify-content: center;
  vertical-align: middle;
  align-items: center;
  padding: 15px 0;
}

.error-element {
  text-align: left;
  color: red;
}
