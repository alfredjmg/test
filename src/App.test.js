import React from 'react';
import App from './App';
import { create } from 'react-test-renderer'

describe('Testing react Test',()=>{
    test('testing App', () => {
    let tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})