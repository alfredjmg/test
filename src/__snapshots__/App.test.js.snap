// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing app button 1`] = `
<div
  className="App"
>
  <div
    className="container-fluid"
  >
    <div
      className="header-container"
    >
      <form
        className="row input-header"
        onSubmit={[Function]}
      >
        <div
          className="col-5"
        >
          <label
            className="visually-hidden"
            htmlFor="inputText"
          >
            Insert text
          </label>
          <input
            className="form-control "
            id="inputText"
            name="inputText"
            onChange={[Function]}
            placeholder="Insert text"
            type="text"
          />
        </div>
        <div
          className="col-auto"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Send
          </button>
        </div>
      </form>
    </div>
    <div
      className="content-container"
    >
      <div
        className="content row"
      >
        <div
          className="title-content col-12 col-md-2"
        >
          Results:
        </div>
        <div
          className="col-12 col-md-10"
        />
        <div
          className="col-12 col-md-2"
        />
        <div
          className="col-12 col-md-8"
        >
          <label
            className="visually-hidden"
            htmlFor="exampleFormControlInput1"
          />
          <input
            className="form-control"
            disabled={true}
            id="exampleFormControlInput1"
            placeholder="First text"
            type="email"
            value=""
          />
        </div>
        <div
          className="col-12 col-md-2"
        />
        <div
          className="col-12 col-md-2"
        />
        <div
          className="palindrome-text col-12 col-md-8"
        >
          Palindrome: 
          <span
            id="palindrome-answer"
          >
            
          </span>
        </div>
        <div
          className="col-12 col-md-2"
        />
      </div>
    </div>
  </div>
</div>
`;
